libs/auth-api-lib/src/lib/delegations/delegations-incoming-custom.service.ts

```diff
@@ -151,6 +178,117 @@ export class DelegationsIncomingCustomService {
     })
   }
 
+  async findAllAvailableGeneralMandate(
+    user: User,
+    clientAllowedApiScopes: ApiScopeInfo[],
+    requireApiScopes: boolean,
+  ): Promise<MergedDelegationDTO[]> {
+    const customApiScopes = clientAllowedApiScopes.filter((s) =>
+      s.supportedDelegationTypes?.some(
+        (dt) => dt.delegationType === AuthDelegationType.GeneralMandate,
+      ),
+    )
+
+    if (requireApiScopes && !(customApiScopes && customApiScopes.length > 0)) {
+      return []
+    }
+
+    const { delegations, fromNameInfo } =
+      await this.findAllIncomingGeneralMandates({
+        validity: DelegationValidity.INCLUDE_FUTURE,
+        nationalId: user.nationalId,
+      })
+
+    const mergedDelegationDTOs = uniqBy(
+      delegations.map((d) =>
+        d.toMergedDTO([AuthDelegationType.GeneralMandate]),
+      ),
+      'fromNationalId',
+    )
+
+    return mergedDelegationDTOs.map((d) => {
+      const person = this.getPersonByNationalId(fromNameInfo, d.fromNationalId)
+
+      return {
+        ...d,
+        fromName: person?.name ?? d.fromName ?? UNKNOWN_NAME,
+      } as MergedDelegationDTO
+    })
+  }
+
+  private async findAllIncomingGeneralMandates(
+    {
+      nationalId,
+      validity,
+    }: FindAllValidIncomingOptions & {
+      validity: DelegationValidity
+    },
+    useMaster = false,
+  ): Promise<{ delegations: Delegation[]; fromNameInfo: FromNameInfo[] }> {
+    const whereOptions = getScopeValidityWhereClause(validity)
```

Comment:
We are not using the `validFrom` fields for GeneralMandates so this becomes a bit weird. As `findAllValidGeneralMandates` uses `INCLUDE_FUTURE` which sounds strange as it should return only valid `NOW`. But you are doing that to work   around the `validFrom` check introduced by this helper.

So I would suggest to not use this helper here, and just arrange the `whereOptions` here to handle `ALL` and `NOW`